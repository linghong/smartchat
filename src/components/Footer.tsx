const Footer = () => {
  return (
    <footer className="py-2 text-center opacity-40 text-sm">
      AI-generated responses may be inaccurate. Verify critical information.
    </footer>
  );
};

export default Footer;
