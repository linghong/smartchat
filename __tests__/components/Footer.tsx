import React from 'react'
import { render, screen } from '@testing-library/react'
import Footer from '@/src/components/Footer'

describe('Footer component', () => {
  it('renders the footer and checks the text content', () => {
    render(<Footer />)
    const footerElement = screen.getByRole('contentinfo')
    expect(footerElement).toBeInTheDocument()
    expect(footerElement).toHaveClass('py-3 text-center opacity-40')
    expect(footerElement.textContent).toBe(
      'Chatbot responses are AI-generated and may be inaccurate, so always verify critical information.',
    )
  })

  it('matches the snapshot', () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
