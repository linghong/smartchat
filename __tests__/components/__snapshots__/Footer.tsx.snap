// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="py-3 text-center opacity-40"
  >
    Chatbot responses are AI-generated and may be inaccurate, so always verify critical information.
  </footer>
</DocumentFragment>
`;
